include/nbl/video/alloc/SubAllocatedDescriptorSet.h

```diff
@@ -0,0 +1,119 @@
+
+            for (auto& binding : bindings)
+            {
+                SubAllocDescriptorSetRange range;
+                range.offset = allocatableDescriptors;
+                range.binding = binding;
+                // Only bindings with these flags will be allocatable
+                if (binding.createFlags.hasFlags(core::bitflag(IGPUDescriptorSetLayout::SBinding::E_CREATE_FLAGS::ECF_UPDATE_AFTER_BIND_BIT) 
+					| IGPUDescriptorSetLayout::SBinding::E_CREATE_FLAGS::ECF_UPDATE_UNUSED_WHILE_PENDING_BIT 
+					| IGPUDescriptorSetLayout::SBinding::E_CREATE_FLAGS::ECF_PARTIALLY_BOUND_BIT))
+                {
+                    allocatableDescriptors += binding.count;
+                }
+				m_allocatableRanges.push_back(range);
+            }
+
+			m_addressAllocator = AddrAllocator(
+				_reservedAllocator.allocate(AddressAllocator::reserved_size(maxAllocatableAlignment, static_cast<size_type>(allocatableDescriptors), args...), _NBL_SIMD_ALIGNMENT),
+				static_cast<size_type>(0), 0u, maxAllocatableAlignment, static_cast<size_type>(allocatableDescriptors), std::forward<Args>(args)...
+			);
+			m_reservedAllocator = ReservedAllocator(std::move(_reservedAllocator));
+            m_reservedSize = allocatableDescriptors;
+        }
+        // version with default constructed reserved allocator
+        template<typename... Args>
+        explicit inline SubAllocatedDescriptorSet(const std::span<const video::IGPUDescriptorSetLayout::SBinding> bindings, 
+            const value_type maxAllocatableAlignment, Args&&... args) :
+            SubAllocatedDescriptorSet(bindings,ReservedAllocator(),maxAllocatableAlignment,std::forward<Args>(args)...)
+        {
+        }
+        ~SubAllocatedDescriptorSet()
+        {
+            auto ptr = reinterpret_cast<const uint8_t*>(core::address_allocator_traits<AddressAllocator>::getReservedSpacePtr(m_addressAllocator));
+            m_reservedAllocator.deallocate(const_cast<uint8_t*>(ptr),m_reservedSize);
+        }
+
+        // anyone gonna use it?
+        inline const AddressAllocator& getAddressAllocator() const {return m_addressAllocator;}
+
+        //
+        inline ReservedAllocator& getReservedAllocator() {return m_reservedAllocator;}
+
+        // main methods
+
+        //! Warning `outAddresses` needs to be primed with `invalid_value` values, otherwise no allocation happens for elements not equal to `invalid_value`
+        template<typename... Args>
+        inline void multi_allocate(uint32_t count, value_type* outAddresses, const size_type* sizes, const Args&... args)
+        {
+            core::address_allocator_traits<AddressAllocator>::multi_alloc_addr(m_addressAllocator,count,outAddresses,sizes,1,args...);
+        }
+        inline void multi_deallocate(uint32_t count, const size_type* addr, const size_type* sizes)
+        {
+            core::address_allocator_traits<AddressAllocator>::multi_free_addr(m_addressAllocator,count,addr,sizes);
+        }
```

Comment:
the allocations from different bindings should not be interchangeable, so this method should take the binding index as the first parameter (before the count)